packages/data-manager/src/agent/dataManager.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  IAgentPlugin,
+  IDataManager,
+  IDataManagerQueryArgs,
+  IDataManagerClearArgs,
+  IDataManagerDeleteArgs,
+  IDataManagerQueryResult,
+  IDataManagerSaveArgs,
+  IDataManagerSaveResult,
+} from '@veramo/core'
+import { AbstractDataStore } from '../data-store/abstractDataStore'
+
+export class DataManager implements IAgentPlugin {
+  readonly methods: IDataManager = {
+    save: this.save.bind(this),
+    query: this.query.bind(this),
+    delete: this.delete.bind(this),
+    clear: this.clear.bind(this),
+  }
+
+  private stores: Record<string, AbstractDataStore>
+
+  constructor(options: { store: Record<string, AbstractDataStore> }) {
+    this.stores = options.store
+  }
+
+  public async save(args: IDataManagerSaveArgs): Promise<Array<IDataManagerSaveResult>> {
+    const data: unknown = args.data
+    const options = args.options
+    let { store } = args.options
+    if (typeof store === 'string') {
+      store = [store]
+    }
+    const res: IDataManagerSaveResult[] = []
+    for (const storeName of store) {
+      const storePlugin = this.stores[storeName]
+      if (!storePlugin) {
+        throw new Error(`Store plugin ${storeName} not found`)
+      }
+      try {
+        const result = await storePlugin.save({ data, options })
+        res.push({ id: result, store: storeName })
+      } catch (e) {
+        console.log(e)
+      }
+    }
+    return res
+  }
+
+  public async query(args: IDataManagerQueryArgs): Promise<Array<IDataManagerQueryResult>> {
+    const { filter = { type: 'none', filter: {} }, options } = args
+    let store
```

Comment:
seems like there's some essentially duplicate code in these 4 functions. should probably refactor it out to a helper function `getStoresFromArgs`?

Reply:
Great idea. I moved that logic to a separate function and added some additional tests